lib/validate.js

```diff
@@ -8,73 +8,63 @@
   The above copyright notice and this permission notice shall be
   included in all copies or substantial portions of this Source Code Form.
 */
-const Joi = require('@hapi/joi');
+const {
+  refinement,
+  any,
+  nullable,
+  type,
+  partial,
+  array,
+  union,
+  number,
+  boolean,
+  string,
+  func,
+  literal,
+  never,
+  validate
+} = require('superstruct');
 const isPromise = require('is-promise');
 
-// until https://github.com/hapijs/joi/issues/1690 is fixed, we have to have a base of any(). once
-// they get that fixed, we can use object() here, which is what we should be using
-const joi = Joi.extend((base) => {
-  return {
-    base,
-    language: {
-      promise: 'must be a promise-like object'
-    },
-    name: 'any',
-    rules: [
-      {
-        name: 'promise',
-        validate(params, value, state, options) {
-          if (!isPromise(value)) {
-            return this.createError('any.promise', { v: value }, state, options);
-          }
-
-          return value;
-        }
-      }
-    ]
-  };
+const promise = refinement(any(), 'promise-like', (value) => {
+  console.log(value, isPromise(value));
```

Comment:
we'll probably want to delete this one yeah?

Reply:
Missed this one